scaladoc/test/dotty/tools/scaladoc/signatures/SignatureTest.scala

```diff
@@ -89,16 +89,16 @@ abstract class SignatureTest(
       afterKind <- Option(kindMatch.after(0)) // to filter out nulls
       name <- if kind.contains("extension") then Some(signature) // The name of an extension will always be the signature itself
                     else identifierRegex.findFirstMatchIn(afterKind).map(_.group(1))
-    yield name
+    yield name.nn
 
   private def signaturesFromSources(source: Source, kinds: Seq[String]): Seq[SignatureRes] =
     source.getLines.map(_.trim)
       .filterNot(_.isEmpty)
       .filterNot(l => l.startWithAnyOfThese("=",":","{","}", "//") && !l.startsWith("//expected:"))
       .toSeq
       .flatMap {
-        case unexpectedRegex(signature) => findName(signature, kinds).map(Unexpected(_))
-        case expectedRegex(signature) => findName(signature, kinds).map(Expected(_, signature))
+        case unexpectedRegex(signature) => findName(signature.nn, kinds).map(Unexpected(_))
```

Comment:
When pattern matching a regex, we should use `unexpectedRegex(signature: String)` to ensure we match a non-empty string element in the pattern. Otherwise, `signature.nn` can cause NPE in runtime with unexpected string.